language/tools/move-cli/src/sandbox/utils/on_disk_state_view.rs

```diff
@@ -112,10 +113,16 @@ impl OnDiskStateView {
     }
 
     // Events are stored under address/handle creation number
-    fn get_event_path(&self, key: &EventKey) -> PathBuf {
-        let mut path = self.get_addr_path(&key.get_creator_address());
+    fn get_event_path(&self, key: &[u8]) -> PathBuf {
+        // TODO: this is a hacky way to get the account address and creation number from the event key.
```

Comment:
Is this hacky? The Event module is in stdlib not the diem framework

Reply:
It's hacky in the sense that our current definition of events is way too Diem-specific, and what I'm doing here is effectively duplicating the logic in `diem_types::event::EventKey`. We will deal with this once we make (Move) events more generic.

Reply:
I suppose a bit with the events dir + native function. But the layout of the key is still defined in the the Events module which seems okay to me

Reply:
> But the layout of the key is still defined in the the Events module which seems okay to me

Yeah, this is not a problem under the current definition given by the Events module. Just saying that this definition itself is Diem-specific and we'll likely need to change it to something more general, in order to support alternative clients like Trove.